contents/tutorials/ruby-ab-tests.md

```diff
@@ -0,0 +1,278 @@
+<script>
+  document.addEventListener('DOMContentLoaded', function() {
+    posthog.onFeatureFlags(function() {
+      var buttonText = 'No variant';
+      if (posthog.getFeatureFlag('my-cool-experiment') === 'control') {
+        buttonText = 'Control variant';
+      } else if (posthog.getFeatureFlag('my-cool-experiment') === 'test') {
+        buttonText = 'Test variant';
+      }
+      document.getElementById('main-cta').textContent = buttonText;
+    });
+  });
+
+  function handleClick() {
+    console.log(posthog)
+    posthog.capture('home_button_clicked');
+  }
+</script>
+
+<main>
+  <h1>Rails A/B Tests</h1>
+  <button id="main-cta" onclick="handleClick()">Click me</button>
+</main>
+```
+
+Now if you refresh your app, you should see the button text updated to either `Control variant` or `Test variant`. 
+
+### Server-side rendering
+
+Notice that when you refresh the page, the button text flickers between `No variant` and `Control/Test variant`. This is because it takes time for PostHog to load and make the feature flag request.
+
+Server-side rendering is a way to avoid this. This fetches the feature flag before the page loads on the client.
+
+To set this up, we must install and use [PostHog’s Ruby SDK](/libraries/ruby) (because we are making server-side requests). Add `posthog-ruby` to your `Gemfile`:
+
+```Gemfile
+gem "posthog-ruby"
+```
+
+Then `bundle install` to install the gem.
+
+Next, we create an initializer for our PostHog client. Create a new file in `config/initializers` called `posthog.rb`. Then, initialize PostHog in this file using your project API key and host:
+
+```ruby file=config/initializers/posthog.rb
+require 'posthog-ruby'
+
+POSTHOG_CLIENT = PostHog::Client.new({
+  api_key: '<ph_project_api_key>',
+  host: '<ph_instance_address>'
+})
+```
+
+Then, we fetch the feature flag in our controller `app/controllers/pages_controller.rb` and set the button text using the value from it:
+
+```ruby file=app/controllers/pages_controller.rb
+class PagesController < ApplicationController
+  def home
+    @button_text = 'No variant'
+    begin
+      distinct_id = 'placeholder-user-id'
+      enabled_variant = POSTHOG_CLIENT.get_feature_flag('my-cool-experiment', distinct_id)
+      if enabled_variant == 'control'
+        @button_text = 'Control Variant'
+      elsif enabled_variant == 'test'
+        @button_text = 'Test Variant'
+      end
+    rescue => e
+      @button_text = 'Error'
+    end
+  end
+end
+```
+
+Lastly, update our view to use the `@button_text` variable from the controller:
+
+```html file=app/views/pages/home.html.erb
+<script>
+  function handleClick() {
+    console.log(posthog)
+    posthog.capture('home_button_clicked');
+  }
+</script>
+
+<main>
+  <h1>Rails A/B Test</h1>
+  <button id="main-cta" onclick="handleClick()"><%= @button_text %></button>
+</main>
+```
+
+Now, when you refresh the page, the button text is already set when the page loads. 
+
```

Comment:
Since we use both the server and client side here, this A/B test isn't going to work properly because the backend ID is going to be different than the frontend ID. I realize we partially solve this in the next section, but maybe we can add a bit more code to pass the server side ID (as a variable) and call identify on the frontend? 

I don't want to end this section with a broken implementation.

Reply:
Added new section on this